force-app/main/default/lwc/bulkServiceDeliveryUI/serviceDeliveryFieldSets.js

```diff
@@ -0,0 +1,118 @@
+import CONTACT_FIELD from "@salesforce/schema/ServiceDelivery__c.Contact__c";
+import SERVICE_FIELD from "@salesforce/schema/ServiceDelivery__c.Service__c";
+import QUANTITY_FIELD from "@salesforce/schema/ServiceDelivery__c.Quantity__c";
+import PROGRAM_ENGAGEMENT_FIELD from "@salesforce/schema/ServiceDelivery__c.ProgramEngagement__c";
+
+import selectService from "@salesforce/label/c.Select_Service";
+import selectEngagement from "@salesforce/label/c.Select_Program_Engagement";
+
+const SHORT_DATA_TYPES = ["DOUBLE", "INTEGER", "BOOLEAN"];
+const LONG_DATA_TYPES = ["TEXTAREA"];
+
+export class ServiceDeliveryFieldSets {
+    fieldSets = {};
+    currentFieldSetName;
+
+    constructor(data) {
+        this.setFieldSets(data);
+    }
+
+    setFieldSets(fieldSets) {
+        for (const [apiName, fieldSet] of Object.entries(fieldSets)) {
+            this.configureFieldSet(apiName, fieldSet);
+        }
+    }
+
+    getCurrentFieldSet() {
+        return this.fieldSets[this.currentFieldSetName];
+    }
+
+    isFieldInCurrentFieldSet(fieldApiName) {
+        return this.isFieldInFieldSet(this.getCurrentFieldSet(), fieldApiName);
+    }
+
+    isFieldInFieldSet(fieldSet, fieldApiName) {
+        let field = fieldSet.find(member => member.apiName === fieldApiName);
+
+        return field !== undefined;
+    }
+
+    hasContactField(fieldSet) {
+        let contactField = fieldSet.find(
+            member => member.apiName === CONTACT_FIELD.fieldApiName
+        );
+
+        return contactField !== undefined;
```

Comment:
```suggestion
        return this.isFieldInFieldSet(fieldSet, CONTACT_FIELD.fieldApiName);
```
Why not use isFieldInFieldSet instead of replicating the find logic? This would apply below too.